torchgeo/samplers/utils.py

```diff
@@ -47,11 +53,17 @@ def get_random_bounding_box(
 
     width = (bounds.maxx - bounds.minx - t_size[1]) // res
     minx = random.randrange(int(width)) * res + bounds.minx
-    maxx = minx + t_size[1]
+    if sample_mode == SIZE_IN_CRS_UNITS:
```

Comment:
I think it may be more reliable to do something like:
```python
if sample_mode == SIZE_IN_PIXELS:
    t_size[0] *= res
    t_size[1] *= res
```
Then the code will be the same for the rest of the function.